nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+            key["collapsed_history"] = make_hash(key)
+            cls.insert1(key, skip_duplicates=True)
+
+        def get_prev_key(self, key):
+            if key["collapsed_history"]:
+                prev_key = self & {"collapsed_history": key["collapsed_history"]}
+                return prev_key.proj(
+                    "model_fn",
+                    "model_hash",
+                    "dataset_fn",
+                    "dataset_hash",
+                    "trainer_fn",
+                    "trainer_hash",
+                    "seed",
+                    collapsed_history="prev_collapsed_history",
+                    next_collapsed_history="collapsed_history",
+                )
+            else:  # no history yet (i.e. transfer step 0)
+                return self.proj() - self  # return something empty
+
+    def _transfer_recipe(self, transfer_step):
+        """
+        Combines multiple transfer recipes and their resitrictions as specified by post_restr attribute.
+        The combination is transfer-step-specific, meaning only the recipes relevant for a specific transfer step would be combined.
+
+        Combining recipes are pretty easy and the user does not need to interact with this method directly. Below is an example:
+        Let us assume you have two recipe tables: TrainerRecipe and ModelRecipe, the you can attach all these recipes to your
+        TransferTrainedModel table as follow:
+
+        ``` Python
+            TransferTrainedModel.transfer_recipe = [TrainerRecipe, ModelRecipe]
+        ```
+
+        The rest (combining the recipes and their restrictions) is taken care of by this method.
+
+        Args:
+            transfer_step (int): table population trasnfer step.
+
+        Returns:
+            string or datajoint AndList: A single or combined restriction of one or multiple recipes, respectively.
+        """
+
+        if isinstance(self.transfer_recipe, list):
+            # get the recipes that have an entry for a specific transfer step
+            transfer_recipe = []
+            for tr in self.transfer_recipe:
+                # check if an entry exists for a specific transfer step in the recipe
+                if tr & f"transfer_step = {transfer_step}":
+                    # if it exists add that entry to the list of recipes (relevant for a specific transfer step)
+                    transfer_recipe.append(tr & f"transfer_step = {transfer_step}")
+            if not transfer_recipe:
+                return self.proj() - self  # return something empty
+            # join all the recipes (and their post_restr)
+            joined = transfer_recipe[0]
+            if len(transfer_recipe) > 1:
+                for t in transfer_recipe[1:]:
+                    joined *= t  # all combination of recipes
+                joined.post_restr = dj.AndList([recipe.post_restr for recipe in self.transfer_recipe])
+            return joined
+        else:
+            return self.transfer_recipe
```

Comment:
consider bringing the simpler resolution case earlier in the if-else brancching

Reply:
What do you mean? Check for the negative and return immediately?

Reply:
I am pretty sure that is exactly what @eywalker means. That way you could get rid of one level of indentation and readers don't need to read till the end of the method if all they care about is the simpler case.